libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6/hwdef.dat

```diff
@@ -100,21 +100,23 @@ PD13 I2C4_SDA I2C4
 # order of I2C buses
 I2C_ORDER I2C2 I2C4
 
+NODMA I2C*
+define STM32_I2C_USE_DMA FALSE
 define HAL_I2C_INTERNAL_MASK 0
 
 # PWM channels
-PE9  TIM1_CH1  TIM1 PWM(1)  GPIO(50)
+PE9  TIM1_CH1  TIM1 PWM(1)  GPIO(50)    BIDIR
 PE11 TIM1_CH2  TIM1 PWM(2)  GPIO(51)
-PE13 TIM1_CH3  TIM1 PWM(3)  GPIO(52)
+PE13 TIM1_CH3  TIM1 PWM(3)  GPIO(52)    BIDIR
 PE14 TIM1_CH4  TIM1 PWM(4)  GPIO(53)
-PA15 TIM2_CH1  TIM2 PWM(5)  GPIO(54)
+PA15 TIM2_CH1  TIM2 PWM(5)  GPIO(54)    BIDIR
 PA1  TIM2_CH2  TIM2 PWM(6)  GPIO(55)
-PA2  TIM2_CH3  TIM2 PWM(7)  GPIO(56)
+PA2  TIM2_CH3  TIM2 PWM(7)  GPIO(56)    BIDIR
 PA3  TIM2_CH4  TIM2 PWM(8)  GPIO(57)
-PC6  TIM3_CH1  TIM3 PWM(9)  GPIO(58)
-PA7  TIM3_CH2  TIM3 PWM(10) GPIO(59)
-PB0  TIM3_CH3  TIM3 PWM(11) GPIO(60)
-PB1  TIM3_CH4  TIM3 PWM(12) GPIO(61)
+PC6  TIM3_CH1  TIM3 PWM(9)  GPIO(58)    NODMA
+PA7  TIM3_CH2  TIM3 PWM(10) GPIO(59)    NODMA
```

Comment:
I don't think you should do this - it only requires a single DMA channel to support regular dshot on all four outputs since they share the same timer

Reply:
Do you mean we can use DMA and BDSHOT on PWM1~12 channels?
As shown below.
# PWM channels
PE9 TIM1_CH1 TIM1 PWM(1) GPIO(50) BIDIR
PE11 TIM1_CH2 TIM1 PWM(2) GPIO(51)
PE13 TIM1_CH3 TIM1 PWM(3) GPIO(52) BIDIR
PE14 TIM1_CH4 TIM1 PWM(4) GPIO(53)
PA15 TIM2_CH1 TIM2 PWM(5) GPIO(54) BIDIR
PA1 TIM2_CH2 TIM2 PWM(6) GPIO(55)
PA2 TIM2_CH3 TIM2 PWM(7) GPIO(56) BIDIR
PA3 TIM2_CH4 TIM2 PWM(8) GPIO(57)
PC6 TIM3_CH1 TIM3 PWM(9) GPIO(58) BIDIR
PA7 TIM3_CH2 TIM3 PWM(10) GPIO(59)
PB0 TIM3_CH3 TIM3 PWM(11) GPIO(60) BIDIR
PB1 TIM3_CH4 TIM3 PWM(12) GPIO(61)
PD14 TIM4_CH3 TIM4 PWM(13) GPIO(62) NODMA
PD15 TIM4_CH4 TIM4 PWM(14) GPIO(63) NODMA

Reply:
Liek this:
```
PE9 TIM1_CH1 TIM1 PWM(1) GPIO(50) BIDIR
PE11 TIM1_CH2 TIM1 PWM(2) GPIO(51)
PE13 TIM1_CH3 TIM1 PWM(3) GPIO(52) BIDIR
PE14 TIM1_CH4 TIM1 PWM(4) GPIO(53)
PA15 TIM2_CH1 TIM2 PWM(5) GPIO(54) BIDIR
PA1 TIM2_CH2 TIM2 PWM(6) GPIO(55)
PA2 TIM2_CH3 TIM2 PWM(7) GPIO(56) BIDIR
PA3 TIM2_CH4 TIM2 PWM(8) GPIO(57)
PC6 TIM3_CH1 TIM3 PWM(9) GPIO(58)
PA7 TIM3_CH2 TIM3 PWM(10) GPIO(59)
PB0 TIM3_CH3 TIM3 PWM(11) GPIO(60)
PB1 TIM3_CH4 TIM3 PWM(12) GPIO(61)
PD14 TIM4_CH3 TIM4 PWM(13) GPIO(62) NODMA
PD15 TIM4_CH4 TIM4 PWM(14) GPIO(63) NODMA
```
Then you have bi-directional dshot PWM1-8, regular dshot 9-12, PWM 14-15

Reply:
Thanks, I already get it.